webui/helpers_test.go

```diff
@@ -0,0 +1,141 @@
+package webui
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/contribsys/faktory/client"
+)
+
+type testJob struct {
+	name     string
+	job      *client.Job
+	expected string
+}
+
+func makeActiveJob(jobType string) *client.Job {
+	var payload = []byte(fmt.Sprintf(`
+{
+  "jid": "bb2a34025e7ce72a064d10d6",
+  "queue": "default",
+  "jobtype": "ActiveJob::QueueAdapters::FaktoryAdapter::JobWrapper",
+  "args": [
+    {
+      "arguments": [
+        1, 2, 3
+      ],
+      "enqueued_at": "2023-12-20T20:36:09.423631911Z",
+      "exception_executions": {},
+      "executions": 0,
+      "job_class": "%[1]s",
+      "job_id": "b24332fc-49bb-4128-8c4c-230c47abea91",
+      "locale": "en",
+      "priority": null,
+      "provider_job_id": "bb2a34025e7ce72a064d10d6",
+      "queue_name": "default",
+      "scheduled_at": null,
+      "timezone": "Pacific Time (US & Canada)"
+    }
+  ],
+  "created_at": "2023-12-20T20:36:09.423723003Z",
+  "enqueued_at": "2023-12-20T20:36:09.423725007Z",
+  "retry": 3,
+  "custom": {
+    "wrapped": "%[1]s"
+  }
+}
+`, jobType))
+	job := &client.Job{}
+	err := json.Unmarshal(payload, job)
+	if err != nil {
+		panic(err)
+	}
+	return job
+}
+
+func makeActionMailerJob(jobType string, mailerClass string, mailerMethod string) *client.Job {
+	var payload = []byte(fmt.Sprintf(`
+{
```

Comment:
This is the Job payload from a Rails 7.1 app. It seems to match the class in https://github.com/sidekiq/sidekiq/blob/73c150d0430a8394cadb5cd49218895b113613a0/test/api_test.rb#L34C135-L34C135, so I labeled it as Rails 6.x further down. I don't know if the specific fields are exactly the same as they would be in Rails 6, but for the purposes of getting the job type it doesn't matter.